Feature/issue643 improve spanish translation (#645)
OWASP/owasp-masvs: Document-es/0x15-V8-Resiliency_Against_Reverse_Engineering_Requirements.md

Los controles de esta sección deben aplicarse según sea necesario, basándose en una evaluación de los riesgos causados por la manipulación no autorizada de la aplicación y/o la ingeniería inversa del código. Sugerimos consultar el documento de OWASP ["Riesgos técnicos de la Ingeniería Inversa y Modificaciones de Código no Autorizadas"](https://wiki.owasp.org/index.php/Technical_Risks_of_Reverse_Engineering_and_Unauthorized_Code_Modification) para obtener una lista de los riesgos de negocio, así como las amenazas técnicas asociadas.

**Tenga en cuenta que los controles de software nunca deben utilizarse como reemplazo de los controles de seguridad. Los controles listados en MASVR-R buscan añadir controles de protección adicionales y específicos contra las amenazas a las aplicaciones que también cumplen con los requerimientos de seguridad del MASVS.**

Se aplican las siguientes consideraciones:

> 1. Debe definirse un modelo de amenaza que defiende claramente las amenazas del lado del cliente. Además, debe especificarse el grado de protección que debe proporcionar el sistema. Por ejemplo, un objetivo podría ser forzar a los autores de malware pensado para obtener información de la aplicación, a que tengan que invertir importantes esfuerzos para realizar la ingeniería inversa.

Reviewer: "defienda" was already right. Could you please correct it? Thank you!

Author: I missunderstood then the phrase 
>> we'll merge it soon and you can update yours with it: https://github.com/OWASP/owasp-masvs/pull/646"

I have undo the merge, but was as a normal commit that i think will cause merge problem later :/
I you need anything (maybe a new branch/PR without intermediate commits) just tell me.

Reviewer: Thanks for undoing the merge that helps a lot.

## Objetivo de Control

> En esta sección se cubren las protecciones recomendadas para aplicaciones que manejan o brindan acceso a información o funcionalidades sensibles. La falta de estos controles no generan vulnerabilidades sino que, están pensados para incrementar la resistencia contra la ingeniería inversa de la aplicación, dificultando al adversario el acceso a los datos o el entendimiento del modo de ejecución de la aplicación.

Reviewer: ```suggestion
En esta sección se cubren las protecciones recomendadas para aplicaciones que manejan o dan acceso a información o funcionalidades sensibles. La falta de estos controles no implica necesariamente que la aplicación contenga vulnerabilidades. Estos controles están pensados para incrementar la resistencia contra la ingeniería inversa, dificultando al adversario el acceso a los datos o el entendimiento del funcionamiento interno de la aplicación.
```

> Los controles de esta sección deben aplicarse según sea necesario, basándose en una evaluación de los riesgos causados por la manipulación no autorizada de la aplicación y/o la ingeniería inversa del código. Sugerimos consultar el documento de OWASP "Ingeniería Inversa - Amenazas de la Ingeniería Inversa de OWASP" (vea las referencias a continuación) para obtener una lista de los riesgos del negocio, así como las amenazas técnicas asociadas.

Reviewer: ```suggestion
Los controles de esta sección deben aplicarse según sea necesario, basándose en una evaluación de los riesgos causados por la manipulación no autorizada de la aplicación y/o la ingeniería inversa del código. Sugerimos consultar el documento de OWASP ["Riesgos técnicos de la Ingeniería Inversa y Modificaciones de Código no Autorizadas"](https://wiki.owasp.org/index.php/Technical_Risks_of_Reverse_Engineering_and_Unauthorized_Code_Modification) para obtener una lista de los riesgos de negocio, así como las amenazas técnicas asociadas.
```

> Para que los controles de la lista siguiente sean eficaces, la aplicación debe cumplir todos los requerimientos del nivel MASVS-L1 (es decir, deben existir sólidos controles de seguridad), así como todos los requerimientos anteriores en V8. Por ejemplo, los controles de ofuscación listados en la sección "Impedir la comprensión" (8.11 y 8.12) deben combinarse con el "Impedir el análisis dinámico y la manipulación" (8.1 - 8.9) y "Asociación del dispositivo" (8.10).

Reviewer: ```suggestion
Para que los controles de la lista siguiente sean eficaces, la aplicación debe cumplir todos los requerimientos del nivel MASVS-L1 (es decir, deben existir sólidos controles de seguridad), así como todos los requerimientos de V8. Por ejemplo, los controles de ofuscación listados en la sección ["Impedir la comprensión"](#impedir-la-comprensión) (8.11 y 8.12) deben combinarse con los de ["Impedir el análisis dinámico y la manipulación"](#impedir-el-análisis-dinámico-y-la-manipulación) (8.1 - 8.9) y ["Asociación del dispositivo"](#asociación-del-dispositivo) (8.10).
```

Author: The new links seems a great idea to me! 👏 

The "anteriores" I think is much better explained before, due without it, you just seems to say you need to achieve all V8, no just de previous from you want to implement. and without it, it will violate the rule of the PR's
"In case of a translation: have you made sure that the translation from EN is as close to the EN document as possible?"

"los de" seems great too.

Reviewer: Thank you for being so considerate of avoiding any violations of the translation and keeping the original meaning. I really appreciate that.

Let me investigate this specific case and I'll come back to you, this is a bit weird, even in the English version.. something doesn't really click for me.

Reviewer: Hi @Sulfkain, I've missed this message unfortunately. Sorry about that.

I'm checking and I can't find any message or mention, do you have a link?

However, please notice that the OWASP Mobile Top 10 is a completely separate project from the OWASP MASVS and MSTG. It has its own leaders which you may want to contact instead:

https://owasp.org/www-project-mobile-top-10/

Milan Singh Thakur: milan@owasp.org
Jonathan Carter: jonathan.carter@owasp.org

**Tenga en cuenta que los controles de software nunca deben utilizarse como reemplazo de los controles de seguridad. Los controles listados en MASVR-R buscan añadir controles de protección adicionales y específicos contra las amenazas a las aplicaciones que también cumplen con los requerimientos de seguridad del MASVS.**

Se aplican las siguientes consideraciones:

> 1. Debe definirse un modelo de amenaza que defiende claramente las amenazas del lado del cliente. Además, debe especificarse el grado de protección que debe proporcionar el sistema. Por ejemplo, un objetivo podría ser forzar a los autores de malware pensado para obtener información de la aplicación, a que tengan que invertir importantes esfuerzos para realizar la ingeniería inversa.

Reviewer: ```suggestion
1. Debe definirse un modelo de amenaza que defienda claramente las amenazas del lado del cliente. Además, debe especificarse el grado de protección que debe proporcionar el sistema. Por ejemplo, un objetivo podría ser forzar a los autores de malware pensado para obtener información de la aplicación, a que tengan que invertir importantes esfuerzos para realizar la ingeniería inversa.
```

> 1. Debe definirse un modelo de amenaza que defienda claramente las amenazas del lado del cliente. Además, debe especificarse el grado de protección que debe proporcionar el sistema. Por ejemplo, un objetivo podría ser obligar a los autores que dirigen malware para obtener información de la aplicación, a que tengan que invertir importantes esfuerzos para realizar la ingeniería inversa.

Reviewer: ```suggestion
1. Debe definirse un modelo de amenaza que defienda claramente las amenazas del lado del cliente. Además, debe especificarse el grado de protección que debe proporcionar el sistema. Por ejemplo, un objetivo podría ser forzar a los autores de malware pensado para obtener información de la aplicación, a que tengan que invertir importantes esfuerzos para realizar la ingeniería inversa.
```

2. El modelo de amenaza debe ser sensato. Por ejemplo, ocultar una clave criptográfica en una implementación de caja blanca es un problema si el atacante puede simplemente utilizar la aplicación como un todo.

> 3. La eficacia de la protección siempre debe ser verificada por un experto con experiencia en el testeo y uso de tipos específicos de anti-manipulación y ofuscación (ver también los capítulos "Ingeniería inversa" y "Evaluación de protecciones del software" en la Guía de Pruebas de Seguridad Móvil).

Reviewer: ```suggestion
3. La eficacia de la protección siempre debe ser verificada por un experto con experiencia en el testeo y uso de tipos específicos de anti-manipulación y ofuscación (ver también el capítulo ["Ingeniería Inversa y Manipulación de Aplicaciones Móviles"](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x04c-Tampering-and-Reverse-Engineering.md) en la Guía de Pruebas de Seguridad Móvil (OWASP MSTG).
```

Author: It lefts one ")" to close the "(ver también....)"

Maybe replace "(ver también.... (OWASP MSTG)" for:

La eficacia de la protección siempre debe ser verificada por un experto con experiencia en el testeo y uso de tipos específicos de anti-manipulación y ofuscación.
_Ver también el capítulo ["Ingeniería Inversa y Manipulación de Aplicaciones Móviles"](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x04c-Tampering-and-Reverse-Engineering.md) en la Guía de Pruebas de Seguridad Móvil (OWASP MSTG)._

We quit () and replace with a full complete phrase anex to the first one.

Reviewer: Nicely seen and I like your suggestion. I've corrected it, you may click now on "commit suggestion".
